src/data.py

```diff
@@ -227,6 +228,107 @@ def __call__(self, sample):
         return (sample - sample.mean()) / (sample.std() + 1e-8)
 
 
+class SlidingWindowNormalize:
+    """ Based on Sliding window augmentations of
+    https://github.com/cchinchristopherj/Right-Whale-Convolutional-Neural-Network/blob/master/whale_cnn.py
+        Translated to torch
+        Has 50/50 chance of activating H sliding window or V sliding window
+
+        Must come after spectrogram and AmplitudeToDB
+    """
+
+    def __init__(self, sr: float, n_fft: int, lower_cutoff: float = 50, norm=True,
+                 inner_ratio: float = 0.06, outer_ratio: float = 0.5):
+        self.sr = sr
+        self.n_fft = n_fft
+        self.lower_cutoff = lower_cutoff
+        self.norm = norm
+        self.inner_ratio = inner_ratio
+        self.outer_ratio = outer_ratio
+
+    # slidingWindowV Function from: https://github.com/nmkridler/moby2/blob/master/metrics.py
+    def slidingWindowV(self, P):
+        ''' slidingWindowV Method
+                Enhance the contrast vertically (along frequency dimension)
+
+                Args:
+                    P: 2-D numpy array image
+                    inner: int length of inner exclusion region (to calculate local mean)
+                    outer: int length of the window (to calculate overall mean)
+                    maxM: int size of the output image in the y-dimension
+                    norm: boolean indicating whether or not to cut off extreme values
+                Returns:
+                    Q: 2-D numpy array image with vertically-enhanced contrast
+
+        '''
+        min_f_ind = int((self.lower_cutoff / (self.sr / 2)) * self.n_fft)
+        Q = P.cpu().clone().numpy()
+        Q_shape = Q.shape
+        Q = Q.squeeze()
+        if self.norm:
+            # Cut off extreme values
+            mval, sval = np.mean(Q[min_f_ind:, :]), np.std(Q[min_f_ind:, :])
+            fact_ = 1.5
+            Q[Q > mval + fact_ * sval] = mval + fact_ * sval
+            Q[Q < mval - fact_ * sval] = mval - fact_ * sval
+            Q[:min_f_ind, :] = mval
+        # Set up the local mean window
+        wInner = np.ones(int(self.inner_ratio * Q.shape[0]))
+        # Set up the overall mean window
+        wOuter = np.ones(int(self.outer_ratio * Q.shape[0]))
+        # Remove overall mean and local mean using np.convolve
+        for i in range(Q.shape[1]):
+            Q[:, i] = Q[:, i] - (np.convolve(Q[:, i], wOuter, 'same') - np.convolve(Q[:, i], wInner, 'same')) / (
+                        wOuter.shape[0] - wInner.shape[0])
+        Q[Q < 0] = 0.
+        return torch.from_numpy(Q).reshape(Q_shape)
+
+    # slidingWindowH Function from: https://github.com/nmkridler/moby2/blob/master/metrics.py
+    def slidingWindowH(self, P):
```

Comment:
I think that most of the code here is similar to slidingWindowV, should we unite them to one function? create base function and two smaller ones?